dlpy/image_captioning.py

```diff
@@ -0,0 +1,551 @@
+#!/usr/bin/env python
+# encoding: utf-8
+#
+# Copyright SAS Institute
+#
+#  Licensed under the Apache License, Version 2.0 (the License);
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+from swat import *
+import pandas as pd
+from dlpy.model import *
+from dlpy.applications import *
+import matplotlib.image as mpimg
+from dlpy.utils import DLPyError
+import random
+
+def get_image_features(conn,model,image_table,dense_layer,target='_filename_0'):
+    '''
+    Generate CASTable of image features
+
+    Parameters
+    ----------
+    conn : CAS
+        Specifies the CAS connection object.
+    model: dlpy Model object
+        Specifies CNN model to use for extracting features
+    image_table: imageTable
+        Specifies name of CASTable that contains images to be used for training
+    dense_layer: string
+        Specifies layer from CNN model to extract features from
+    target: string, optional
+        Specifies the name of the column containing the response variable
+        Default: '_filename_0'
+
+    Returns
+    -------
+    CASTable

```

Comment:
check other function, I believe we use ` so that we can link them from the documentation